packages/zowe-explorer/src/trees/shared/SharedInit.ts

```diff
@@ -430,11 +436,22 @@ export class SharedInit {
         });
     }
 
-    public static async setupRemoteWorkspaceFolders(e?: vscode.WorkspaceFoldersChangeEvent): Promise<void> {
+    public static async setupRemoteWorkspaceFolders(e?: vscode.WorkspaceFoldersChangeEvent, profileType?: string): Promise<void> {
         // Perform remote lookup for workspace folders that fit the `zowe-ds` or `zowe-uss` schemes.
-        const newWorkspaces = (e?.added ?? vscode.workspace.workspaceFolders ?? []).filter(
+        let newWorkspaces = (e?.added ?? vscode.workspace.workspaceFolders ?? []).filter(
             (f) => f.uri.scheme === ZoweScheme.DS || f.uri.scheme === ZoweScheme.USS
         );
+
+        const profInfo = Profiles.getInstance();
+        let profileNames: string[] = [];
```

Comment:
(suggestion) With less than 10 profiles, an array has minimal difference than a set, but with several profiles in a team config, you may find `Set` and `Set.has` to be more performant than `Array.includes`:

| Method | Avg. Case | Worst Case |
| --------- | ---------- | ------------- |
| `Set.has` | O(1) | O(n) |
| `Array.includes` | O(n) | O(n) |